pkg/agent/agent.go

```diff
@@ -106,8 +128,78 @@ func (a *Agent) Run(healthCheck *metrics.HealthCheck, enableProfiling bool, bind
 	}()
 
 	<-a.ctx.Done()
+
+	_ = a.DeleteNodeResourceTsp()
 }
 
 func getAgentName(nodeName string) string {
-	return nodeName + "_" + string(uuid.NewUUID())
+	return nodeName + "." + string(uuid.NewUUID())
+}
+
+func (a *Agent) CreateNodeResourceTsp() string {
+	tsp, err := a.craneClient.PredictionV1alpha1().TimeSeriesPredictions(resource.TspNamespace).Get(context.TODO(), a.GenerateNodeResourceTspName(), metav1.GetOptions{})
+	if err == nil {
+		klog.V(4).Infof("Found old tsp %s in namespace %s", a.GenerateNodeResourceTspName(), resource.TspNamespace)
+		err := a.DeleteNodeResourceTsp()
+		if err != nil {
+			klog.Exitf("Delete old tsp %s with error: %v", a.GenerateNodeResourceTspName(), err)
+		}
+	}
+	config, err := a.kubeClient.CoreV1().ConfigMaps(resource.TspNamespace).Get(context.TODO(), "noderesource-tsp-template", metav1.GetOptions{})
+
+	if err != nil {
+		klog.Exitf("Get noderesource tsp configmap noderesource-tsp-template with error: %v", err)
+	}
+
+	if config == nil {
+		klog.Exitf("Can't get noderesource tsp configmap noderesource-tsp-template")
+	}
+
+	spec := v1alpha12.TimeSeriesPredictionSpec{}
+	err = yaml.Unmarshal([]byte(strings.Replace(config.Data["spec"], "{{nodename}}", a.nodeName, -1)), &spec)
```

Comment:
Can here use go-template insteaded?

Reply:
Of course, if there are more parameters that need to be templated.